public/app/features/trails/TrailStore/TrailStore.ts

```diff
@@ -55,9 +57,15 @@ export class TrailStore {
 
     t.history.map((step) => {
       this._loadFromUrl(trail, step.urlValues);
+      const parentIndex = step.parentIndex ?? trail.state.history.state.steps.length - 1;
+      // Set the parent of the next trail step by setting the current step in history.
+      trail.state.history.setState({ currentStep: parentIndex });
       trail.state.history.addTrailStep(trail, step.type);
     });
 
+    const currentStep = t.currentStep ?? trail.state.history.state.steps.length - 1;
+    trail.state.history.setState({ currentStep });
+
```

Comment:
@adrapereira wanted to make sure you saw the changes to persistence before I merge.